Publishing paper to JOSS (#692)
suzil/radis-app: paper/paper.md

---
title: 'Radis app : web app for Radis high-resolution infrared molecular spectra'
tags:
  - python
  - spectroscopy
  - physics
  - web app
  - Radis
  - molecular
  - spectra
authors:
  - name: Arunava Basu
    orcid: 0009-0000-2880-2142
  - name: Susannah Klanecek
    orcid: 0000-0002-3325-6569
affiliations:
date: 19 July 2023
bibliography: paper.bib

# Optional fields if submitting to a AAS journal too, see this blog post:
# https://blog.joss.theoj.org/2018/12/a-new-collaboration-with-aas-publishing
aas-doi: 10.3847/xxxxx <- update this with the DOI from AAS once you know it.
aas-journal: Astrophysical Journal <- The name of the AAS journal.
---
# Summary

**Radis App** is a web application for Radis high-resolution infrared molecular spectra. Instead of writing code, this project aims to create an intuitive user interface (UI).

> It use Radis internally to produce spectrum, and the updated version and Radis algorithm make it incredibly efficient to compute the millions of lines in only a few minutes.

Reviewer: ```suggestion
It uses Radis internally to produce spectrum, and the updated version and Radis algorithm makes it incredibly efficient to compute millions of lines in only a few minutes.
```

Radis app leverages React 18 to offer the user interface, and FastApi on the backend. We are using react-hook-form for the fastest user experience and to maintain performance on the client slide. In the backend, we use FastApi to offer the fastest response.

# Statement of need

> We build  **Radis App** to  address an important need in the field of high-resolution infrared molecular spectroscopy. Traditional approaches for analyzing  molecular spectra often involve complex coding and lack intuitive user interfaces, leading to barriers for researchers and practitioners in effectively utilizing this valuable data.

Reviewer: ```suggestion
We build  **Radis App** to address an important need in the field of high-resolution infrared molecular spectroscopy. Traditional approaches for analyzing  molecular spectra often involve complex coding and lack intuitive user interfaces, leading to barriers for researchers and practitioners in effectively utilizing this valuable data.
```

> The Radis app fills this gap by providing a user-friendly web application that leverages the power of the Radis algorithm. By abstracting  the need for coding, the Radis app enables researchers to efficiently generate high-resolution spectra with ease. This not only saves significant time and effort but also opens up opportunities for a wider range of users to explore and analyze molecular spectra.

Reviewer: ```suggestion
The Radis app fills this gap by providing a user-friendly web application that leverages the power of the Radis algorithm. By abstracting the need for coding, the Radis app enables researchers to efficiently generate high-resolution spectra with ease. This not only saves significant time and effort but also opens up opportunities for a wider range of users to explore and analyze molecular spectra.
```

The ability of the Radis app to plot spectra, export them, and overlay multiple spectra for comparative analysis further enhances its usefulness. This comprehensive set of features caters to the diverse needs of researchers working in various fields, ranging from atmospheric science to chemical engineering.

# App Architecture

- **Radis Architecture -**

![Radis  Architecture](./images/radis_architecture.png)

- **Radis App Architecture -**

![Radis App Architecture](./images/radisapp_architecture.png)

# Features

The Radis app offers several key features that enhance the user experience and facilitate efficient analysis of high-resolution infrared molecular spectra. These features includes -

> 1. **Plotting Spectra**: The app allows users to plot the generated spectra visually. This feature provides a clear and concise representation of the spectral data, enabling researchers to easily analyses and interpret the results. Visualizations can be customized and adjusted to meet specific requirements, enhancing data comprehension and facilitating further analysis.

Reviewer: ```suggestion
1. **Plotting Spectra**: The app allows users to plot the generated spectra visually. This feature provides a clear and concise representation of the spectral data, enabling researchers to easily analyze and interpret the results. Visualizations can be customized and adjusted to meet specific requirements, enhancing data comprehension and facilitating further analysis.
```
